src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userOptional = userService.findByEmail(email);
+        if (userOptional.isEmpty()) {
+            throw new AuthenticationException("User is not found by email " + email);
+        }
+        User user = userOptional.get();
+        String hashedPassword = HashUtil.hashPassword(password, user.getSalt());
+        if (!user.getPassword().equals(hashedPassword)) {
+            throw new AuthenticationException("Password is incorrect for user with email " + email);
+        }
+        return user;
+    }
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
+        try {
```

Comment:
```suggestion
        validateRegisterData(email, password);
        return userService.add(new User(email, password));
```

Validate `if email == null || email.isBlank() || password == null || password.isBlank()` and also `if (userService.findByEmail(email).isPresent())`